tests/zenko_tests/python_tests/zenko_e2e/util/bucket.py

```diff
@@ -35,12 +36,15 @@ def enable_versioning(bucket):
     bucket.Versioning().enable()
 
 
-def get_object_hash(bucket, key, timeout=0, backoff=5, _timestamp=None):
+def get_object_hash(bucket, key, timeout=0, backoff=5, versionid=None, _timestamp=None):
     if _timestamp is None:
         _timestamp = time.time()
+    extra_args = {}
+    if versionid is not None:
+        extra_args['VersionId'] = versionid
     try:
         with tempfile.TemporaryFile() as tfile:
```

Comment:
I just saw the E2E pod became OOM. Might have to use `SpooledTemporaryFile` to limit how much gets spooled in memory. Or just remove the upper memory limit for that pod since typically the test cluster has an abundance of available memory.

Reply:
I'll swap the tempfile to `SpooledTemporaryFile` and bump the limit to 2Gi. Our biggest "file" is only 100MB so we shouldn't be using that much memory.

Reply:
Yeah it's my bad really. I use a 1GB file for the cosmos test and thought the 1.5 would be enough. 2Gi should definitely be enough.